server/user.go

```diff
@@ -44,6 +44,10 @@ func (c *Connection) JiraAccountID() types.ID {
 
 type ConnectionSettings struct {
 	Notifications bool `json:"notifications"`
+
+	// these will take effect if Notifications is true
+	SendNotificationsForAssigned bool `json:"send_notifications_for_assigned"`
+	SendNotificationsForReporter bool `json:"send_notifications_for_reporter"`
```

Comment:
To make things backwards compatible, we can introduce `*bool` fields. This allows us to know if the value should be in its "default" state.

```suggestion
    Notifications bool `json:"notifications"`
	SendNotificationsForMention *bool `json:"send_notifications_for_mention"`
	SendNotificationsForAssignee *bool `json:"send_notifications_for_assignee"`
    SendNotificationsForReporter *bool `json:"send_notifications_for_reporter"`
```

Then we can use this logic to determine the value:

```go
sendForAssigned := false
if settings.SendNotificationsForAssigned != nil {
    sendForAssigned = *settings.SendNotificationsForAssigned
} else {
    sendForAssigned = settings.Notifications
}
```

The user then sees three notification options available when changing their settings. When a user connects, they should have the default settings of all notification values as true.